attachments/simple_engine/renderer_rendering.cpp

```diff
@@ -0,0 +1,948 @@
+                                    pushConstants.roughnessFactor = material->roughness;
+
+                                    // Texture set flags (-1 = no texture)
+                                    pushConstants.baseColorTextureSet = material->albedoTexturePath.empty() ? -1 : 0;
+                                    if (material->useSpecularGlossiness) {
+                                        pushConstants.useSpecGlossWorkflow = 1;
+                                        pushConstants.physicalDescriptorTextureSet = material->specGlossTexturePath.empty() ? -1 : 0;
+                                        pushConstants.glossinessFactor = material->glossinessFactor;
+                                        pushConstants.specularFactor = material->specularFactor;
+                                    } else {
+                                        pushConstants.useSpecGlossWorkflow = 0;
+                                        pushConstants.physicalDescriptorTextureSet = material->metallicRoughnessTexturePath.empty() ? -1 : 0;
+                                    }
+                                    pushConstants.normalTextureSet = material->normalTexturePath.empty() ? -1 : 0;
+                                    pushConstants.occlusionTextureSet = material->occlusionTexturePath.empty() ? -1 : 0;
+                                    pushConstants.emissiveTextureSet = material->emissiveTexturePath.empty() ? -1 : 0;
+
+                                    // Emissive and transmission/IOR
+                                    pushConstants.emissiveFactor = material->emissive;
+                                    pushConstants.emissiveStrength = material->emissiveStrength;
+                                    pushConstants.hasEmissiveStrengthExtension = false; // Material has emissive strength data
+                                    pushConstants.transmissionFactor = material->transmissionFactor;
+                                    pushConstants.ior = material->ior;
+
+                                    // Alpha mask handling
+                                    pushConstants.alphaMask = (material->alphaMode == "MASK") ? 1.0f : 0.0f;
+                                    pushConstants.alphaMaskCutoff = material->alphaCutoff;
+                                }
+                            }
+                        }
+                    }
+                }
+                commandBuffers[currentFrame].pushConstants<MaterialProperties>(**currentLayout, vk::ShaderStageFlagBits::eFragment, 0, { pushConstants });
+                uint32_t instanceCountT = std::max(1u, static_cast<uint32_t>(meshComponent->GetInstanceCount()));
+                commandBuffers[currentFrame].drawIndexed(meshIt->second.indexCount, instanceCountT, 0, 0, 0);
+            }
+        }
+
+        if (imguiSystem) {
+            imguiSystem->Render(commandBuffers[currentFrame], currentFrame);
+        }
+        commandBuffers[currentFrame].endRendering();
+    }
```

Comment:
The opaque and the transparent path seem to be very similar. Maybe worth to introduce a helper function?